src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Projects/PackageReferenceProject.cs

```diff
@@ -22,11 +29,25 @@ namespace NuGet.PackageManagement.VisualStudio
     /// </summary>
     public abstract class PackageReferenceProject : BuildIntegratedNuGetProject
     {
+        private static readonly CacheItemPolicy TransitiveOriginsCacheItemPolicy = new()
+        {
+            SlidingExpiration = ObjectCache.NoSlidingExpiration,
+            AbsoluteExpiration = ObjectCache.InfiniteAbsoluteExpiration,
+        };
+        private static readonly ObjectCache TransitiveOriginsCache = new MemoryCache("TransitiveOriginsCache", new NameValueCollection
```

Comment:
By making this static, I think there might be a logical error as the cache would be shared across different projects.

Reply:
What's the benefit of an object cache over a simple concurrent dictionary?

We're not really dealing with timed expiration here.

Reply:
Benefits of using cache:

- Keeping elements at capacity when memory is low
- Better semantics

Reply:
Made instance variable.

Reply:
I'd argue that a strongly typed dictionary has a better semantics than an ambigious ObjectCache. 

I'd love to hear what others think @NuGet/nuget-client

Reply:
From my understanding intention for `ObjectCache` use in Server env where you deal with infinite/very long stream of small data keep coming and it offers sliding time expiration for old data which less relevant. It looks we used this one already in our PMUI infinite scroll where intention sounds very similar. But here DFS algorithm works in short period and finite so may not fit intention, maybe simple dictionary is better fit.